### Parte 1: Análisis Exploratorio de Datos (EDA)

1. **Carga de los datos:**

   - Deberás cargar los dos conjuntos de datos que te hemos proporcionado para realizar este Laboratorio:

     - `netflix_originals.csv`

     - `netflix_titles.csv`

   Deberás revisar la estructura de cada conjunto de datos utilizando los métodos aprendidos durante la lección.

In [1]:
# Lo primero que hacemos es importar las librerias Panda y Numpy. Y hacemos que se muuestren los anchos de columna.
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None) # para poder visualizar todas las columnas de los DataFrames


### Importamos los csv.

In [2]:
datos_original = pd.read_csv("../datos/netflix_originals.csv", index_col = 0)
datos_original.head(5)


Title                  Genre           Premiere  Runtime  \
0  Enter the Anime            Documentary     August 5, 2019       58   
1      Dark Forces               Thriller    August 21, 2020       81   
2          The App  Science fiction/Drama  December 26, 2019       79   
3   The Open House        Horror thriller   January 19, 2018       94   
4      Kaali Khuhi                Mystery   October 30, 2020       90   

   IMDB Score          Language  
0         2.5  English/Japanese  
1         2.6           Spanish  
2         2.6           Italian  
3         3.2           English  
4         3.4             Hindi

In [3]:
datos_titulos = pd.read_csv("../datos/netflix_titles.csv", index_col = 0)
datos_titulos.head(5)

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

### Ahora vamos a analizar los 2 data frame.

In [4]:
# Miramos la info de datos originales
visualizar_datos = datos_original.info()
visualizar_datos

<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 28.1+ KB


Vemos que tenemos 6 columnas 4 de tipo str, 1 int + 1 float

In [5]:
# Miramos la info de datos titulos
visualizar_titulos = datos_titulos.info()
visualizar_titulos

<class 'pandas.core.frame.DataFrame'>
Index: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 894.5+ KB


En este csv vemos que tenemos 12 columnas que son todas tipo str(object) menos realease year que es tipo int(numerica)

### Ahora realizamos eda numerico de los 2 dataframes.

In [6]:
datos_original.describe().T

count       mean        std  min   25%   50%    75%    max
Runtime     513.0  94.674464  26.648121  4.0  87.0  97.0  108.0  209.0
IMDB Score  513.0   6.210916   0.968850  2.5   5.6   6.3    6.9    9.0

- Podemos tener con exactitud los datos que nos pasan de RunTime y de IMDB Score, con lo que podemos comprobar los valores.
  - En la columna Runtime podemos apreciar que hay bastante similitud entre la media y la mediana que esta en 97 y 94.6. Por lo que ese valor nos da indicios que el tiempo de las peliculas rondan los 90 minutos. 
  - En la columna IMDB Score los datos son un poco mas dispersos ya que solo tenemos ese datos de un porcentaje muy bajo de los datos generales por que solo se han votado las peliculas del dataset originales, por lo que no tenemos mucho más analisis que mostrar.

In [7]:
datos_titulos.describe().T

count         mean       std     min     25%     50%     75%  \
release_year  8807.0  2014.180198  8.819312  1925.0  2013.0  2017.0  2019.0   

                 max  
release_year  2021.0

-  En esta ocasión comprobamos los datos exactos de la columna release_year donde se puede ver las estadisticas de ella.
   -  En año de lanzamiento podemos ver que la media y mediana esta también bastante proxima y con la proximidad de 25% y de 75% de los datos se puede ver que las peliculas fueron lanzadas entre 2014 y 2017.

### Ahora realizamos la misma operación pero con los datos tipo Object

In [8]:
datos_original.describe(include='object').T

count unique                                       top freq
Title      513    513  David Attenborough: A Life on Our Planet    1
Genre      513    106                               Documentary  132
Premiere   513    350                          October 18, 2019    5
Language   513     37                                   English  352

Con el dataset original podemos apreciar que en la columna Title, tenemos las 513 peliculas unicas es un buen dato para ver que no hay duplicados, en Genre hay solo 106 datos unicos y el más repetido es Documentatty con 132 ocasiones. Con Premier son datos bastante vólatiles ya que son fecha de lanzamiento y aparecen 350 valores únicos por lo que no tiene relevancia que haya alguno repetido.Y por último la columna Language que tiene 37 valores únicos que nos dice que el data set 37 lenguajes diferentes y el más repetido es English.

In [9]:
datos_titulos.describe(include='object').T

count unique                                                top  \
show_id      8807   8807                                              s8807   
type         8807      2                                              Movie   
title        8807   8807                                             Zubaan   
director     6173   4528                                      Rajiv Chilaka   
cast         7982   7692                                 David Attenborough   
country      7976    748                                      United States   
date_added   8797   1767                                    January 1, 2020   
rating       8803     17                                              TV-MA   
duration     3994    185                                          2 Seasons   
listed_in    8807    514                       Dramas, International Movies   
description  8807   8775  Paranormal activity at a lush, abandoned prope...   

             freq  
show_id         1  
type         6131  
title           1  
director       19  
cast           19  
country      2818  
date_added    109  
rating       3207  
duration      425  
listed_in     362  
description     4

Vamos a analizar los valores object del dataset titutos:
- La columna show_id nos muestra 8807 valores unicos, nos da información que no tiene valores duplicados.
- La columna Type Solo tiene 2 valores unicos que son TV Show y Movie, que separa mas peliculas de las series. Donde hay mas del 75% de valores que son pelicula.
- El la columna Title vemos que todos los valores son unicos por lo que nos da a entender que no tenemos valores duplicados.
- En la columna Director es vemos que tenemos 4528 valores únicos por lo que nos da a entender que hay muchos directores y el más repetido es Rajiv Chilaka con 19 peliculas. El resto son valores NaN y podemos cambiar el nombre a Desconocido.
- La columna cast nos muestra una gran cantidad de valores unicos y los restantes son tipo NaN por lo que podemos decidir ponerlo como desconocido.
- En la columna Country tenemos 748 datos unicos que son los diferentes paises que aparecen en la columna, y podemos ver que en más del 30% el valor más repetido osea en pais que mas peliculas tiene es USA.
- La columna Date_added nos muestra cuando se añadio la pelicula a la platarforma, por lo que podemos sacar buenas concluisiones en futuras consultas.
- La columna Rating tiene 17 columnas unicas, la más representativa con bastante diferencia es TV-Ma con 3207 de freq.
- En la columna Duration vemos que hay 185 diferentes tipos de duration de las peliculas en las que vemos que las que tiene 2 seasons son las mas repetidas.
- Listed_in tiene 514 valores unicos, que nos da ha entender que hay muchos tipos de peliculas en la columna. El valor más repetido es Dramas, Internacional Movies.
- En description vemos que casi todas las descripciones son únicas solo 4 tienen la misma repeticion y son peliculas de actividad paranormal.
  
- 

### Analizamos alguna columna para ver sus valores

En este caso seleccionamos de tipo int.

In [47]:
datos_original.select_dtypes(include=['int']).T


0    1    2    3    4    5    6    7    8    9    11   12   13   14   \
Runtime   58   81   79   94   90  147  112  149   73  139  112   97  101   90   

         15   16   17   18   19   21   22   23   24   25   26   27   28   29   \
Runtime   25  144  115  102  100   97   99  120  105   89   97  107   99   95   

         30   31   32   34   35   36   37   38   39   40   41   42   43   44   \
Runtime   37   89   83   85   88   86  105   95   80    4   93  100  106   97   

         45   46   47   48   49   50   51   52   53   54   55   56   57   58   \
Runtime  106  103   80  101  119   80   89   94   93   96  113   86  100  102   

         59   60   61   62   63   64   66   67   69   70   71   72   73   74   \
Runtime   86  104   88   97  105   90   10  106   94  112  117   70   81  103   

         75   76   77   78   80   81   82   83   84   85   86   87   88   89   \
Runtime   94   98  131   87  112  102   99  116   92   83   97  112  116  102   

         90   91   92   93   94   95   96   99   100  101  102  103  104  105  \
Runtime  121   95   93   92   98   85   78  101  119  101  112   96  114   97   

         106  107  108  109  110  111  112  113  114  115  117  118  119  120  \
Runtime  101   56  100   94   94   21  104   63  100  126   86  142  108  102   

         121  122  123  124  125  126  127  128  129  130  131  132  133  134  \
Runtime   92   83   94   92  103   89   95  103   97  125   91   90   49   98   

         135  136  137  138  139  140  141  142  143  144  145  146  147  148  \
Runtime  107  118   94  103   86   83   97   99  104  102   89   98  125   89   

         149  150  151  152  153  154  155  156  157  158  159  161  162  163  \
Runtime   94   94   85   34   86   91  108   95  100   90  112  124  107   92   

         164  165  166  167  168  169  170  171  172  173  174  175  176  177  \
Runtime  121   90  104   92  114   98   95   78  107   92   95   96  118  105   

         178  179  181  182  183  184  185  187  188  189  190  191  192  193  \
Runtime   94   52   95  100   97   89  102  131   89   93  111   75  120   96   

         194  195  196  198  200  201  202  203  204  205  206  207  208  211  \
Runtime  148   94  107  104  111   83   53   95  132   98  108  106  104   90   

         212  213  214  215  216  217  218  219  220  221  222  223  224  226  \
Runtime   97  113  123  115  105  101  102  122  128   82  102   80   94  104   

         227  228  229  230  231  232  233  234  235  236  237  238  239  240  \
Runtime   85   84  103   95  100   89  116  118   96  114   99   87   92   90   

         242  243  244  245  246  247  248  250  251  252  253  254  255  256  \
Runtime   96   93   42   92  106  151  101   90  115   92   72  139   98   92   

         257  258  259  261  262  263  264  265  266  267  268  269  270  271  \
Runtime  111   98  104   88  102   30   99  129   87  117  104  111  107   97   

         273  274  275  276  277  279  280  281  282  283  284  285  286  287  \
Runtime   90  101   72   83  118  113   86  108   85  102  151   98   87  120   

         288  290  291  292  294  295  297  298  299  300  301  302  303  304  \
Runtime  134   85  103   94   88   37  102  100   96  113   16  119   97   81   

         305  306  307  308  310  311  312  313  314  315  316  317  318  319  \
Runtime   98   95   96  107   87  101   97  103   90  125  116   99   82   97   

         320  322  323  324  325  327  328  329  330  331  332  333  334  335  \
Runtime   28   93   74   60    9  100  155   55  123  103   93  119   40  120   

         336  337  338  339  340  341  342  343  344  346  347  348  349  350  \
Runtime   98  104   91   95  101  105   83   91   83   86  124   94  102  118   

         351  352  353  354  355  356  357  358  359  360  361  362  363  364  \
Runtime   79  134  126   98   89   58   83  136   98  114   99  130   32  104   

         365  366  367  369  371  372  373  374  375  376  377  378  379  381  \
Runtime  142   9

Ahora lo hacemos de tipo object

In [11]:
datos_original.select_dtypes(include=['object'])


Title                  Genre  \
0                                Enter the Anime            Documentary   
1                                    Dark Forces               Thriller   
2                                        The App  Science fiction/Drama   
3                                 The Open House        Horror thriller   
4                                    Kaali Khuhi                Mystery   
..                                           ...                    ...   
576                       Cuba and the Cameraman            Documentary   
577                       Dancing with the Birds            Documentary   
580  Winter on Fire: Ukraine's Fight for Freedom            Documentary   
581                      Springsteen on Broadway           One-man show   
583     David Attenborough: A Life on Our Planet            Documentary   

              Premiere                  Language  
0       August 5, 2019          English/Japanese  
1      August 21, 2020                   Spanish  
2    December 26, 2019                   Italian  
3     January 19, 2018                   English  
4     October 30, 2020                     Hindi  
..                 ...                       ...  
576  November 24, 2017                   English  
577   October 23, 2019                   English  
580    October 9, 2015  English/Ukranian/Russian  
581  December 16, 2018                   English  
583    October 4, 2020                   English  

[513 rows x 4 columns]

- En esta ocasión vemos valores de la columna Genre para identificarlos.

In [12]:
datos_original["Genre"].value_counts

<bound method IndexOpsMixin.value_counts of 0                Documentary
1                   Thriller
2      Science fiction/Drama
3            Horror thriller
4                    Mystery
               ...          
576              Documentary
577              Documentary
580              Documentary
581             One-man show
583              Documentary
Name: Genre, Length: 513, dtype: object>

- Del data set titulos decido ver los valores de Listen_in

In [13]:
datos_titulos["listed_in"].value_counts

<bound method IndexOpsMixin.value_counts of 0                                           Documentaries
1         International TV Shows, TV Dramas, TV Mysteries
2       Crime TV Shows, International TV Shows, TV Act...
3                                  Docuseries, Reality TV
4       International TV Shows, Romantic TV Shows, TV ...
                              ...                        
8802                       Cult Movies, Dramas, Thrillers
8803               Kids' TV, Korean TV Shows, TV Comedies
8804                              Comedies, Horror Movies
8805                   Children & Family Movies, Comedies
8806       Dramas, International Movies, Music & Musicals
Name: listed_in, Length: 8807, dtype: object>

In [48]:
datos_original.sample(10)

Title                  Genre  \
24                       Thunder Force       Superhero-Comedy   
388                           Upstarts                  Drama   
148                           Good Sam                  Drama   
169                       Class of '83                  Drama   
485                       Ladies First            Documentary   
512        Secrets of the Saqqara Tomb            Documentary   
323          Altered Carbon: Resleeved  Anime/Science fiction   
234                             Sergio                 Biopic   
73   Handsome: A Netflix Mystery Movie                 Comedy   
204                           The Prom                Musical   

              Premiere  Runtime  IMDB Score        Language  original  
24       April 9, 2021      105         4.4         English  original  
388   October 18, 2019      112         6.7           Hindi  original  
148       May 16, 2019       89         5.7         English  original  
169    August 21, 2020       98         5.8           Hindi  original  
485      March 8, 2018       39         7.2   English/Hindi  original  
512   October 28, 2020      114         7.3  English/Arabic  original  
323     March 19, 2020       74         6.5        Japanese  original  
234     April 17, 2020      118         6.1         English  original  
73         May 5, 2017       81         5.2         English  original  
204  December 11, 2020      132         5.9         English  original

In [49]:
datos_titulos.sample(10)


show_id     type                                     Title  \
5983   s5984    Movie  100 Years: One Woman's Fight for Justice   
5491   s5492    Movie                                    Sahara   
864     s865    Movie                             Taxi No. 9211   
2285   s2286    Movie                                   Players   
1182   s1183  TV Show               Formula 1: Drive to Survive   
1897   s1898    Movie                               Serious Men   
1743   s1744    Movie                                    MOTHER   
396     s397    Movie                             Blood Red Sky   
1117   s1118    Movie                  Ibrahim a Fate to Define   
4479   s4480  TV Show                                Great News   

                                               director  \
5983                                      Melinda Janko   
5491                                        Pierre Coré   
864                                       Milan Luthria   
2285  Abbas Alibhai Burmawalla, Mastan Alibhai Burma...   
1182                                        Desconocido   
1897                                      Sudhir Mishra   
1743                                     Tatsushi Omori   
396                                     Peter Thorwarth   
1117                                       Lina Al Abed   
4479                                        Desconocido   

                                                   cast  \
5983                                        Desconocido   
5491  Robert Naylor, Angela Galuppo, Daniel Brochu, ...   
864   Nana Patekar, John Abraham, Sameera Reddy, Son...   
2285  Vinod Khanna, Abhishek Bachchan, Bobby Deol, S...   
1182                                        Desconocido   
1897  Nawazuddin Siddiqui, Nasser, Aakshath Das, Ind...   
1743        Masami Nagasawa, Sadao Abe, Daiken Okudaira   
396   Peri Baumeister, Alexander Scheer, Kais Setti,...   
1117                                        Desconocido   
4479  Briga Heelan, Andrea Martin, Adam Campbell, Jo...   

                                 country        date_added  release_year  \
5983                       United States    March 20, 2018          2016   
5491                      France, Canada      May 12, 2017          2017   
864                                  NaN      May 20, 2021          2006   
2285                               India      July 5, 2020          2012   
1182                      United Kingdom    March 19, 2021          2021   
1897                               India   October 2, 2020          2020   
1743                               Japan  November 3, 2020          2020   
396                                  NaN     July 23, 2021          2021   
1117  Lebanon, Palestine, Denmark, Qatar     April 1, 2021          2019   
4479                       United States  October 25, 2018          2017   

     rating   duration                                          listed_in  \
5983  TV-14     75 min                                      Documentaries   
5491  TV-Y7        NaN                           Children & Family Movies   
864   TV-14    116 min            Dramas, International Movies, Thrillers   
2285  TV-14    162 min           Action & Adventure, International Movies   
1182  TV-MA        NaN                                         Docuseries   
1897  TV-MA        NaN             Comedies, Dramas, International Movies   
1743  TV-14    127 min                       Dramas, International Movies   
396   TV-MA    123 min  Action & Adventure, Horror Movies, Internation...   
1117  TV-PG     75 min                Documentaries, International Movies   
4479  TV-14  2 Seasons                                        TV Comedies   

                                            description  
5983  This documentary chronicles Elouise Cobell's l...  
5491  When his new love is captured by an evil snake...  
864   When an arrogant scion hails his taxi, a cabbi...  
2285  Learning of a fortune in gold being secretly

In [51]:
datos_original.info()


<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, 0 to 583
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
 6   original    513 non-null    object 
dtypes: float64(1), int64(1), object(5)
memory usage: 48.2+ KB


### Miro si hay duplicados y nulos

In [17]:
# Aqui miramos si hay datos duplicados por filas
datos_duplicados_original = (datos_original.duplicated().sum())
print(datos_duplicados_original)

0


In [18]:
datos_duplicados_titulos = (datos_titulos.duplicated().sum())
datos_duplicados_titulos

np.int64(0)

In [19]:
# Vamos a ver los valores nulos de cada data frame.
datos_nulos_original = (datos_original.isnull().sum())
datos_nulos_original

Title         0
Genre         0
Premiere      0
Runtime       0
IMDB Score    0
Language      0
dtype: int64

In [20]:
datos_original.isnull().T

0      1      2      3      4      5      6      7      8    \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              9      11     12     13     14     15     16     17     18   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              19     21     22     23     24     25     26     27     28   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              29     30     31     32     34     35     36     37     38   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              39     40     41     42     43     44     45     46     47   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              48     49     50     51     52     53     54     55     56   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              57     58     59     60     61     62     63     64     66   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Premiere    False  False  False  False  False  False  False  False  False   
Runtime     False  False  False  False  False  False  False  False  False   
IMDB Score  False  False  False  False  False  False  False  False  False   
Language    False  False  False  False  False  False  False  False  False   

              67     69     70     71     72     73     74     75     76   \
Title       False  False  False  False  False  False  False  False  False   
Genre       False  False  False  False  False  False  False  False  False   
Pre

En el dataframe datos originales vemos que no tenemos datos nulos.

En el dataframe de titulos tenemos 4 columnas que tienen muchos datos nulos. 

In [21]:
datos_titulos.isna().sum()


show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration        4813
listed_in          0
description        0
dtype: int64

### Calculo el porcentaje de los datos nulos para ver el impacto en el dataset.

In [22]:
print((datos_titulos.isnull().sum() / len(datos_titulos)) * 100)



show_id          0.000000
type             0.000000
title            0.000000
director        29.908028
cast             9.367549
country          9.435676
date_added       0.113546
release_year     0.000000
rating           0.045418
duration        54.649710
listed_in        0.000000
description      0.000000
dtype: float64


 Decido ponerle a director y cast, los Nan como "Desconocido". 

In [23]:
datos_titulos["director"].fillna('Desconocido', inplace=True)
datos_titulos["cast"].fillna('Desconocido', inplace=True)
datos_titulos.head(2)


C:\Users\User\AppData\Local\Temp\ipykernel_4620\1562111278.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  datos_titulos["director"].fillna('Desconocido', inplace=True)
C:\Users\User\AppData\Local\Temp\ipykernel_4620\1562111278.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

show_id     type                 title         director  \
0      s1    Movie  Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show         Blood & Water      Desconocido   

                                                cast        country  \
0                                        Desconocido  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   

                                         listed_in  \
0                                    Documentaries   
1  International TV Shows, TV Dramas, TV Mysteries   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...

### Realizo la unión de los csv y lo hago con la columna title, y Title. 

Antes cambio el nombre de title del data set titulos para que no me duplique las columnas

In [24]:
datos_titulos = datos_titulos.rename(columns={"title" : "Title"})

Y compruebo que todo este ok.

In [25]:
datos_titulos

show_id     type                  Title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water      Desconocido   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans      Desconocido   
4         s5  TV Show           Kota Factory      Desconocido   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb      Desconocido   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                           Desconocido  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                           Desconocido            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                        Desconocido            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA        NaN   
3     September 24, 2021          2021  TV-MA        NaN   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R        NaN   
8805    January 11, 2020          2006     PG        NaN   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

Ahora creo la columna original para saber las series que son originales de netflix

In [26]:
datos_original["original"] = "original" 

In [27]:
# Compruebo que todo esta ok
datos_original

Title                  Genre  \
0                                Enter the Anime            Documentary   
1                                    Dark Forces               Thriller   
2                                        The App  Science fiction/Drama   
3                                 The Open House        Horror thriller   
4                                    Kaali Khuhi                Mystery   
..                                           ...                    ...   
576                       Cuba and the Cameraman            Documentary   
577                       Dancing with the Birds            Documentary   
580  Winter on Fire: Ukraine's Fight for Freedom            Documentary   
581                      Springsteen on Broadway           One-man show   
583     David Attenborough: A Life on Our Planet            Documentary   

              Premiere  Runtime  IMDB Score                  Language  \
0       August 5, 2019       58         2.5          English/Japanese   
1      August 21, 2020       81         2.6                   Spanish   
2    December 26, 2019       79         2.6                   Italian   
3     January 19, 2018       94         3.2                   English   
4     October 30, 2020       90         3.4                     Hindi   
..                 ...      ...         ...                       ...   
576  November 24, 2017      114         8.3                   English   
577   October 23, 2019       51         8.3                   English   
580    October 9, 2015       91         8.4  English/Ukranian/Russian   
581  December 16, 2018      153         8.5                   English   
583    October 4, 2020       83         9.0                   English   

     original  
0    original  
1    original  
2    original  
3    original  
4    original  
..        ...  
576  original  
577  original  
580  original  
581  original  
583  original  

[513 rows x 7 columns]

### Realizo la union de las 2 data base por que ya lo tengo preparado

- Decido unirlas por Title por que es la columna que compartimos

In [28]:
datos_netflix = datos_titulos.merge(datos_original, how='left',  on = "Title")
datos_netflix


show_id     type                  Title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water      Desconocido   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans      Desconocido   
4         s5  TV Show           Kota Factory      Desconocido   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb      Desconocido   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                           Desconocido  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                           Desconocido            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                        Desconocido            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA        NaN   
3     September 24, 2021          2021  TV-MA        NaN   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R        NaN   
8805    January 11, 2020          2006     PG        NaN   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description        Genre  \
0     As her father nears the end of his life, filmm...  Documentary   
1     After crossing paths at a party, a Cape Town t...          NaN   
2     To protect his family from a powerful drug lor...          NaN   
3     Feuds, flirtations and toilet talk go down amo...          NaN   
4     In a city of coaching centers known to train I...          NaN   
...                                                 ...          ...   
8802  A political cartoonist, a crime reporter and a...          NaN   
8803  While living alone in a spooky town, a young g...          NaN   
8804  Looking to survive in a world taken over by zo...          NaN   
8805  Dragged from civilian life, a former superhero...          NaN   
8806  A scrappy but poor boy worms his way into a ty...          NaN   

             Premiere  Runtime  IMDB Sco

In [29]:
datos_netflix.shape

(8807, 18)

### Visualizo datos netflix para comprobar que todo esta ok

In [52]:
datos_netflix.sample(10)

show_id     type                      Title  \
485     s486    Movie                Lethal Love   
2728   s2729    Movie  Humpty Sharma Ki Dulhania   
2683   s2684    Movie                   Mischief   
6427   s6428    Movie              Casino Royale   
8578   s8579    Movie                     Thithi   
5500   s5501    Movie      English Babu Desi Mem   
3857   s3858    Movie                Last Breath   
918     s919  TV Show             Lava Ka Dhaava   
5455   s5456    Movie               Maya Memsaab   
1982   s1983    Movie                   Anaamika   

                              director  \
485                     Avi Federgreen   
2728                  Shashank Khaitan   
2683                       Emcy Joseph   
6427                   Martin Campbell   
8578                        Raam Reddy   
5500                   Praveen Nischol   
3857  Alex Parkinson, Richard da Costa   
918                        Desconocido   
5455                       Ketan Mehta   
1982                    Sekhar Kammula   

                                                   cast  \
485   David Pinard, Joelle Farrow, Torri Higginson, ...   
2728  Varun Dhawan, Alia Bhatt, Ashutosh Rana, Sidha...   
2683  Suraj Venjaramoodu, Soubin Shahir, Surabhi Lak...   
6427  Daniel Craig, Judi Dench, Eva Green, Jeffrey W...   
8578  Thammegowda S., Channegowda, Abhishek H.N., Po...   
5500  Shah Rukh Khan, Sonali Bendre, Rajeshwari, Sae...   
3857                                        Desconocido   
918                       Jaaved Jaaferi, Rutledge Wood   
5455  Deepa Sahi, Farooq Shaikh, Raj Babbar, Shah Ru...   
1982  Nayantara, Vaibhav Reddy, Pasupathy, Harshvard...   

                                                country          date_added  \
485                                                 NaN        July 9, 2021   
2728                                              India       April 1, 2020   
2683                                              India      April 15, 2020   
6427  United Kingdom, Czech Republic, United States,...     August 31, 2020   
8578                               India, United States       March 9, 2017   
5500                                              India         May 1, 2017   
3857                    United Kingdom, Belgium, Sweden         May 6, 2019   
918                                               India         May 5, 2021   
5455                      France, United Kingdom, India        June 1, 2017   
1982                                              India  September 17, 2020   

      release_year rating duration  \
485           2021  TV-14      NaN   
2728          2014  TV-14  131 min   
2683          2019  TV-PG  120 min   
6427          2006  PG-13  145 min   
8578          2015  TV-MA  123 min   
5500          1996  TV-14  163 min   
3857          2019  TV-MA      NaN   
918           2021  TV-PG      NaN   
5455          1993  TV-14  122 min   
1982          2014  TV-14  130 min   

                                             listed_in  \
485                                          Thrillers   
2728   Comedies, International Movies, Romantic Movies   
2683              Comedies, Dramas, Independent Movies   
6427                                Action & Adventure   
8578  Dramas, Independent Movies, International Movies   
5500            Comedies, Dramas, International Movies   
3857               Documentaries, International Movies   
918                                         Reality TV   
5455    Dramas, International Movies, Music & Musicals   
1982                   International Movies, Thrillers   

                                            description Genre Premiere  \
485   After a business owner starts receiving online...   NaN      NaN   
2728  A small-town girl heads to Delhi to find a des...   NaN      NaN   
2683  Mistaking the sleeping passenger on his train ...   NaN      NaN   
6427  Daniel Craig makes his debut as the newly mint...   NaN      NaN   
8578  In a South Indi

In [53]:
### Guardo el archivo ya unido
datos_netflix.to_csv("../datos/eda_netflix.csv")

### COMPROBACIONES 

In [54]:
datos_netflix.sample(10)

show_id     type                                             Title  \
8187   s8188    Movie                            The American President   
7207   s7208    Movie                                         King Jack   
8505   s8506    Movie                                          The Show   
1271   s1272  TV Show                                  The Last Bastion   
6338   s6339    Movie                         Blue Is the Warmest Color   
3539   s3540  TV Show  Loo Loo Kids: Johny & Friends Musical Adventures   
5809   s5810    Movie                                              XOXO   
8557   s8558  TV Show                                     The West Wing   
914     s915    Movie                                            Unrest   
8368   s8369  TV Show                                     The Irish Mob   

                 director                                               cast  \
8187           Rob Reiner  Michael Douglas, Annette Bening, Martin Sheen,...   
7207       Felix Thompson  Charlie Plummer, Cory Nichols, Christian Madse...   
8505   Giancarlo Esposito  Josh Duhamel, Giancarlo Esposito, Famke Jansse...   
1271          Desconocido  Giovanni Arce, Mayella Lloclla, Rodrigo Palaci...   
6338  Abdellatif Kechiche  Léa Seydoux, Adèle Exarchopoulos, Salim Kechio...   
3539          Desconocido                                        Desconocido   
5809    Christopher Louie  Sarah Hyland, Graham Phillips, Chris D'Elia, H...   
8557          Desconocido  Martin Sheen, Rob Lowe, Allison Janney, John S...   
914         Jennifer Brea  Jennifer Brea, Omar Wasow, Jessica Taylor-Bearman   
8368          Desconocido                                    Mickey Spillane   

                            country          date_added  release_year rating  \
8187                  United States     January 1, 2021          1995  PG-13   
7207                  United States        July 1, 2016          2015     NR   
8505                  United States        June 1, 2020          2017  TV-MA   
1271                            NaN   February 25, 2021          2018  TV-MA   
6338         France, Belgium, Spain     August 26, 2016          2013  NC-17   
3539                            NaN   September 1, 2019          2016   TV-Y   
5809                  United States     August 26, 2016          2016  TV-MA   
8557                  United States   December 25, 2015          2005  TV-14   
914   United Kingdom, United States         May 7, 2021          2017  TV-MA   
8368                            NaN    December 1, 2016          2008  TV-MA   

       duration                                          listed_in  \
8187    113 min                  Comedies, Dramas, Romantic Movies   
7207     81 min                         Dramas, Independent Movies   
8505        NaN                                             Dramas   
1271        NaN  International TV Shows, Spanish-Language TV Sh...   
6338    180 min   Dramas, Independent Movies, International Movies   
3539        NaN                                           Kids' TV   
5809        NaN                           Dramas, Music & Musicals   
8557  7 Seasons                                          TV Dramas   
914         NaN                                      Documentaries   
8368        NaN                         Crime TV Shows, Docuseries   

                                            description  Genre  \
8187  The widowed president strikes up a romance wit...    NaN   
7207  Already coping with a bully, a mercurial older...    NaN   
8505  A disillusioned reality TV host and an idealis...    NaN   
1271  The Robles family, born and raised in the Amer...    NaN   
6338  Determined to fall in love, 15-year-old Adele ...    NaN   
3539  Music-loving baby Johny goes on dance-filled a...    NaN   
5809  The lives of a budding DJ, his loyal pal, a ho...  Drama   
8557  This powerful political epic chronicles the tr...    NaN   
914   In this heart-wrenching documentary, a 28-year

     - ¿Cuántos títulos en `netflix_titles.csv` son producciones originales de Netflix?


In [33]:
# Titulos originales Netflix
originales = datos_netflix["original"].notnull().sum()
originales

np.int64(513)

In [34]:
# Titulos no originales Netflix
no_originales = datos_netflix["original"].isnull().sum()
no_originales

np.int64(8294)

     - ¿Hay diferencias significativas en la puntuación de IMDB entre títulos originales y no originales?


No podemos evaluar dicha pregunta por que el dato IMDB solo es un dato que existe en titulos originales y no esta en titulos no originales.

### Parte 3: Uso de `loc` e `iloc`


   - Selecciona todas las filas donde el tipo de contenido sea “Movie”.


In [35]:
movies_datos_netflix = datos_netflix.loc[datos_netflix['type'] == 'Movie']
movies_datos_netflix

show_id   type                             Title  \
0         s1  Movie              Dick Johnson Is Dead   
6         s7  Movie  My Little Pony: A New Generation   
7         s8  Movie                           Sankofa   
9        s10  Movie                      The Starling   
12       s13  Movie                      Je Suis Karl   
...      ...    ...                               ...   
8801   s8802  Movie                           Zinzana   
8802   s8803  Movie                            Zodiac   
8804   s8805  Movie                        Zombieland   
8805   s8806  Movie                              Zoom   
8806   s8807  Movie                            Zubaan   

                           director  \
0                   Kirsten Johnson   
6     Robert Cullen, José Luis Ucha   
7                      Haile Gerima   
9                    Theodore Melfi   
12              Christian Schwochow   
...                             ...   
8801                Majid Al Ansari   
8802                  David Fincher   
8804                Ruben Fleischer   
8805                   Peter Hewitt   
8806                    Mozez Singh   

                                                   cast  \
0                                           Desconocido   
6     Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7     Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
...                                                 ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                country          date_added  \
0                                         United States  September 25, 2021   
6                                                   NaN  September 24, 2021   
7     United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
9                                         United States  September 24, 2021   
12                              Germany, Czech Republic  September 23, 2021   
...                                                 ...                 ...   
8801                       United Arab Emirates, Jordan       March 9, 2016   
8802                                      United States   November 20, 2019   
8804                                      United States    November 1, 2019   
8805                                      United States    January 11, 2020   
8806                                              India       March 2, 2019   

      release_year rating duration  \
0             2020  PG-13   90 min   
6             2021     PG      NaN   
7             1993  TV-MA  125 min   
9             2021  PG-13  104 min   
12            2021  TV-MA  127 min   
...            ...    ...      ...   
8801          2015  TV-MA      NaN   
8802          2007      R  158 min   
8804          2009      R      NaN   
8805          2006     PG      NaN   
8806          2015  TV-14  111 min   

                                             listed_in  \
0                                        Documentaries   
6                             Children & Family Movies   
7     Dramas, Independent Movies, International Movies   
9                                     Comedies, Dramas   
12                        Dramas, International Movies   
...                                                ...   
8801           Dramas, International Movies, Thrillers   
8802                    Cult Movies, Dramas, Thrillers   
8804                           Comedies, Horror Movies   
8805                Children & Family Movies, Comedies   
8806    Dramas, International Movies, Music & Musicals   

                                       

   - Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.


In [36]:
titulo_duracion_netflix = datos_netflix.loc[datos_netflix['type'] == 'TV Show', ['Title', 'duration']]
titulo_duracion_netflix

Title   duration
1             Blood & Water  2 Seasons
2                 Ganglands        NaN
3     Jailbirds New Orleans        NaN
4              Kota Factory  2 Seasons
5             Midnight Mass        NaN
...                     ...        ...
8795        Yu-Gi-Oh! Arc-V  2 Seasons
8796             Yunus Emre  2 Seasons
8797              Zak Storm        NaN
8800     Zindagi Gulzar Hai        NaN
8803            Zombie Dumb  2 Seasons

[2676 rows x 2 columns]

   - Selecciona todas las filas donde el país sea “United States”.


In [37]:
filas_United_States = datos_netflix.loc[datos_netflix['country'] == 'United States']
filas_United_States

show_id     type                 Title         director  \
0         s1    Movie  Dick Johnson Is Dead  Kirsten Johnson   
9        s10    Movie          The Starling   Theodore Melfi   
15       s16  TV Show     Dear White People      Desconocido   
27       s28    Movie             Grown Ups     Dennis Dugan   
28       s29    Movie            Dark Skies    Scott Stewart   
...      ...      ...                   ...              ...   
8791   s8792    Movie           Young Adult    Jason Reitman   
8793   s8794    Movie  Yours, Mine and Ours     Raja Gosnell   
8802   s8803    Movie                Zodiac    David Fincher   
8804   s8805    Movie            Zombieland  Ruben Fleischer   
8805   s8806    Movie                  Zoom     Peter Hewitt   

                                                   cast        country  \
0                                           Desconocido  United States   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...  United States   
15    Logan Browning, Brandon P. Bell, DeRon Horton,...  United States   
27    Adam Sandler, Kevin James, Chris Rock, David S...  United States   
28    Keri Russell, Josh Hamilton, J.K. Simmons, Dak...  United States   
...                                                 ...            ...   
8791  Charlize Theron, Patton Oswalt, Patrick Wilson...  United States   
8793  Dennis Quaid, Rene Russo, Sean Faris, Katija P...  United States   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   

              date_added  release_year rating duration  \
0     September 25, 2021          2020  PG-13   90 min   
9     September 24, 2021          2021  PG-13  104 min   
15    September 22, 2021          2021  TV-MA      NaN   
27    September 20, 2021          2010  PG-13      NaN   
28    September 19, 2021          2013  PG-13      NaN   
...                  ...           ...    ...      ...   
8791   November 20, 2019          2011      R      NaN   
8793   November 20, 2019          2005     PG      NaN   
8802   November 20, 2019          2007      R  158 min   
8804    November 1, 2019          2009      R      NaN   
8805    January 11, 2020          2006     PG      NaN   

                                 listed_in  \
0                            Documentaries   
9                         Comedies, Dramas   
15                  TV Comedies, TV Dramas   
27                                Comedies   
28         Horror Movies, Sci-Fi & Fantasy   
...                                    ...   
8791  Comedies, Dramas, Independent Movies   
8793    Children & Family Movies, Comedies   
8802        Cult Movies, Dramas, Thrillers   
8804               Comedies, Horror Movies   
8805    Children & Family Movies, Comedies   

                                            description        Genre  \
0     As her father nears the end of his life, filmm...  Documentary   
9     A woman adjusting to life after a loss contend...          NaN   
15    Students of color navigate the daily slights a...          NaN   
27    Mourning the loss of their beloved junior high...          NaN   
28    A family’s idyllic suburban life shatters when...          NaN   
...                                                 ...          ...   
8791  When a divorced writer gets a letter from an o...          NaN   
8793  When a father of eight and a mother of 10 prep...          NaN   
8802  A political cartoonist, a crime reporter and a...          NaN   
8804  Looking to survive in a world taken over by zo...          NaN   
8805  Dragged from civilian life, a former superhero...          NaN   

             Premiere  Runtime  IMDB Score Language  original  
0     October 2, 2020     90.0         7.5  English  original  
9                 NaN      NaN         NaN      NaN       NaN  
15                NaN      NaN         N

   - Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.


In [38]:
tipo_movie = datos_netflix.loc[datos_netflix['type'] == 'Movie', ['Title', 'duration']]
tipo_movie

Title duration
0                 Dick Johnson Is Dead   90 min
6     My Little Pony: A New Generation      NaN
7                              Sankofa  125 min
9                         The Starling  104 min
12                        Je Suis Karl  127 min
...                                ...      ...
8801                           Zinzana      NaN
8802                            Zodiac  158 min
8804                        Zombieland      NaN
8805                              Zoom      NaN
8806                            Zubaan  111 min

[6131 rows x 2 columns]

   - Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.


In [39]:
producciones_2018 = datos_netflix.loc[datos_netflix['release_year'] == 2018, ['Title', 'listed_in']]
producciones_2018

Title  \
21                               Resurrection: Ertugrul   
37                                          Angry Birds   
73                                         King of Boys   
94                                            Show Dogs   
96    If I Leave Here Tomorrow: A Film About Lynyrd ...   
...                                                 ...   
8732                                      White Chamber   
8743                                           Wildlife   
8774                                          يوم الدين   
8775                                    Yeh Meri Family   
8803                                        Zombie Dumb   

                                              listed_in  
21    International TV Shows, TV Action & Adventure,...  
37                                Kids' TV, TV Comedies  
73                         Dramas, International Movies  
94                   Children & Family Movies, Comedies  
96                      Documentaries, Music & Musicals  
...                                                 ...  
8732                        Sci-Fi & Fantasy, Thrillers  
8743                         Dramas, Independent Movies  
8774   Dramas, Independent Movies, International Movies  
8775                International TV Shows, TV Comedies  
8803             Kids' TV, Korean TV Shows, TV Comedies  

[1147 rows x 2 columns]

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).


In [40]:
director_desconocido = datos_netflix.loc[datos_netflix['director'] == 'Desconocido', ['Title']]
director_desconocido

Title
1                           Blood & Water
3                   Jailbirds New Orleans
4                            Kota Factory
10    Vendetta: Truth, Lies and The Mafia
14        Crime Stories: India Detectives
...                                   ...
8795                      Yu-Gi-Oh! Arc-V
8796                           Yunus Emre
8797                            Zak Storm
8800                   Zindagi Gulzar Hai
8803                          Zombie Dumb

[2634 rows x 1 columns]

2. **Acceso a datos específicos con `iloc`:**


   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.


In [41]:
cinco_filas = datos_netflix.iloc[:5,[datos_netflix.columns.get_loc('Title'), datos_netflix.columns.get_loc('director'), datos_netflix.columns.get_loc ('country')]]
cinco_filas

Title         director        country
0   Dick Johnson Is Dead  Kirsten Johnson  United States
1          Blood & Water      Desconocido   South Africa
2              Ganglands  Julien Leclercq            NaN
3  Jailbirds New Orleans      Desconocido            NaN
4           Kota Factory      Desconocido          India

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.



In [42]:
ultimas_filas = datos_netflix.iloc[-5:,[datos_netflix.columns.get_loc('Title'), datos_netflix.columns.get_loc('description')]]
ultimas_filas

Title                                        description
8802       Zodiac  A political cartoonist, a crime reporter and a...
8803  Zombie Dumb  While living alone in a spooky town, a young g...
8804   Zombieland  Looking to survive in a world taken over by zo...
8805         Zoom  Dragged from civilian life, a former superhero...
8806       Zubaan  A scrappy but poor boy worms his way into a ty...

   - Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 


In [43]:
primeras_diez = datos_netflix.iloc[:10, 1:6]
primeras_diez

type                             Title                       director  \
0    Movie              Dick Johnson Is Dead                Kirsten Johnson   
1  TV Show                     Blood & Water                    Desconocido   
2  TV Show                         Ganglands                Julien Leclercq   
3  TV Show             Jailbirds New Orleans                    Desconocido   
4  TV Show                      Kota Factory                    Desconocido   
5  TV Show                     Midnight Mass                  Mike Flanagan   
6    Movie  My Little Pony: A New Generation  Robert Cullen, José Luis Ucha   
7    Movie                           Sankofa                   Haile Gerima   
8  TV Show     The Great British Baking Show                Andy Devonshire   
9    Movie                      The Starling                 Theodore Melfi   

                                                cast  \
0                                        Desconocido   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                        Desconocido   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                             country  
0                                      United States  
1                                       South Africa  
2                                                NaN  
3                                                NaN  
4                                              India  
5                                                NaN  
6                                                NaN  
7  United States, Ghana, Burkina Faso, United Kin...  
8                                     United Kingdom  
9                                      United States

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.


In [44]:
ultimas_siete = datos_netflix.iloc[-7:,[datos_netflix.columns.get_loc('Title'), datos_netflix.columns.get_loc('director'), datos_netflix.columns.get_loc('country')]]
ultimas_siete

Title         director                       country
8800  Zindagi Gulzar Hai      Desconocido                      Pakistan
8801             Zinzana  Majid Al Ansari  United Arab Emirates, Jordan
8802              Zodiac    David Fincher                 United States
8803         Zombie Dumb      Desconocido                           NaN
8804          Zombieland  Ruben Fleischer                 United States
8805                Zoom     Peter Hewitt                 United States
8806              Zubaan      Mozez Singh                         India

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.


In [45]:
filas_entre = datos_netflix.iloc[15:26, 1:5]
filas_entre

type                                              Title  \
15  TV Show                                  Dear White People   
16    Movie  Europe's Most Dangerous Man: Otto Skorzeny in ...   
17  TV Show                                    Falsa identidad   
18    Movie                                          Intrusion   
19  TV Show                                             Jaguar   
20  TV Show    Monsters Inside: The 24 Faces of Billy Milligan   
21  TV Show                             Resurrection: Ertugrul   
22    Movie                                    Avvai Shanmughi   
23    Movie       Go! Go! Cory Carson: Chrissy Takes the Wheel   
24    Movie                                              Jeans   
25  TV Show                               Love on the Spectrum   

                                         director  \
15                                    Desconocido   
16  Pedro de Echave García, Pablo Azorín Williams   
17                                    Desconocido   
18                                     Adam Salky   
19                                    Desconocido   
20                                Olivier Megaton   
21                                    Desconocido   
22                                 K.S. Ravikumar   
23                        Alex Woo, Stanley Moore   
24                                     S. Shankar   
25                                    Desconocido   

                                                 cast  
15  Logan Browning, Brandon P. Bell, DeRon Horton,...  
16                                        Desconocido  
17  Luis Ernesto Franco, Camila Sodi, Sergio Goyri...  
18  Freida Pinto, Logan Marshall-Green, Robert Joh...  
19  Blanca Suárez, Iván Marcos, Óscar Casas, Adriá...  
20                                        Desconocido  
21  Engin Altan Düzyatan, Serdar Gökhan, Hülya Dar...  
22  Kamal Hassan, Meena, Gemini Ganesan, Heera Raj...  
23  Maisie Benson, Paul Killam, Kerry Gudjohnsen, ...  
24  Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...  
25                                   Brooke Satchwell

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.


In [46]:
filas_datos = datos_netflix.iloc[20:31,[datos_netflix.columns.get_loc('Title'), datos_netflix.columns.get_loc('release_year'), datos_netflix.columns.get_loc('rating')]]
filas_datos

Title  release_year rating
20  Monsters Inside: The 24 Faces of Billy Milligan          2021  TV-14
21                           Resurrection: Ertugrul          2018  TV-14
22                                  Avvai Shanmughi          1996  TV-PG
23     Go! Go! Cory Carson: Chrissy Takes the Wheel          2021   TV-Y
24                                            Jeans          1998  TV-14
25                             Love on the Spectrum          2021  TV-14
26                                   Minsara Kanavu          1997  TV-PG
27                                        Grown Ups          2010  PG-13
28                                       Dark Skies          2013  PG-13
29                                         Paranoia          2013  PG-13
30                                  Ankahi Kahaniya          2021  TV-14